### ISSUE TITLE
Mixed ckeditor versions

### ISSUE THREAD

**Issue by interjinn** [Mediawiki-wysiwyg/WYSIWYG-CKeditor#26]
Hi, I'm trying to track down a weird bug when switching back and forth between source and WYSIWYG mode in the editor:

```
TypeError: a is undefined (ckeditor.js:781)
```

This kills the editor and occurs with my test page on the second or third switch (it's not consistent). To track it down I tried switching to using the uncompiled ckeditor. This took some work due to the missing plugins textselection and texttransform in _source/plugins. Once I did get it working though I noticed that the CKEditor version had fallen back to 3.6. Is it possible to have this updated to be in synch with the minified CKEditor? Another wish would be to have a version of the compiled ckeditor.js that was built with the --leave-js-unminified flag set. This would make tracking down CKEditor issues a lot simpler.

Thanks,
Rob.

**Comment by riparap** [comment]
I am a newbie with .js / .php / browser programming so I must admit that I am not expert on this field. Please be patient if I ask trivial things :)

When I updated Ckeditor from 3.6 to 4.x.x, I was wondering what was the _source directory used for. Because I could not figure this out, none of the files of _source directory have been modified so they are from original unmodified bundle of wysiwyg which had version 3.6 of CKeditor (=branch "master" of this bundle of wysiwyg).

When I downloaded CKeditor 4.x.x, I have used option "Optimized" of CKeditor download page. I did not use option "Source (Big N'Slow)", because I did not know what these source files would be used for at that point. So.... will this be solved simply if I download CKeditor with that "Source...." option and merge all missing/modified files into our bundle of wysiwyg editor? ....or am I on the wrong tracks here.

About option ckeditor.js and option "--leave-js-unminified":
At this moment I do not know how to do this, because I have not done any compilations with CKeditor earlier myself. Would you be able to give a little more information about how this can be accomplished :)

**Comment by interjinn** [comment]
No worries, you've been doing an excellent job with WYSIWYG that I can see. I think to resolve the issue you would need to determine what files are not in CKEditor 4.4 since they should be removed from the git repository after overwiting the existing files. I can look into doing it if you want via my fork and you could merge it in if you're comfortable. I'll have to investigate the --leave-js-unminified option a bit more since I'm not sure it's as simple as compiling from the CKEditor dev branch. Thanks!

**Comment by riparap** [comment]
At first I could not find "Source (Big N’Slow)" files of CKeditor 4.4.4 so I updated version of CKEditor to 4.4.5 because this update is rather easy thing to do. I hope 4.4.5 works, because I did not do much testing with it. 

After this I noticed that source files of CKeditor can be found from github development repository. I completely replaced ckeditor_source directory with downloaded github source files of CKeditor 4.4.5 and copied following subdirectories from directory ckeditor\plugins to directory ckeditor_source\plugins: mediawiki, mwtemplate, smwqueryinterface, smwrichmedia, smwrule, smwtoolbar, smwwebservice, textselection, texttransform. 

I was able to run wysiwyg using source files of CKeditor, but I had to make modified config.js for it (comments how I did it are in _source/_ReadMe-ckeditor_source.txt + _source/config.js). 

I can also use CKBuilder to compile files of _source directory. Resulting ckeditor seems to work so I think files of _source directory should be ok.

Based on this thread https://github.com/ckeditor/ckeditor-releases/issues/11 it is possible to use option --leave-js-unminified with command build.sh (of CKBuilder). That parameter must be manually added into ckbuilder.jar command. I am just wondering if this option is used, does it have some kind of performance effects in same way as running plain source files of CKeditor?
